lib/media/track/localtrackpublication.js

```diff
@@ -1,42 +1,35 @@
 'use strict';
 
-const buildLogLevels = require('../../util').buildLogLevels;
-const Log = require('../../util/log');
-const DEFAULT_LOG_LEVEL = require('../../util/constants').DEFAULT_LOG_LEVEL;
-let nInstances = 0;
+const TrackPublication = require('./trackpublication');
 
 /**
  * A {@link LocalTrackPublication} is a {@link LocalTrack} that has been
  * published to a {@link Room}.
  * @property {Track.Kind} kind - kind of the published {@link LocalTrack}
  * @property {LocalTrack} track - the {@link LocalTrack}
- * @property {string} trackName - the {@link LocalTrack}'s name
- * @property {Track.SID} trackSid - SID assigned to the published
- *   {@link LocalTrack}
  */
-class LocalTrackPublication {
+class LocalTrackPublication extends TrackPublication {
   /**
    * Construct a {@link LocalTrackPublication}.
-   * @param {Track.SID} sid - SID assigned to the published {@link LocalTrack}
+   * @param {Track.SID} trackSid - SID assigned to the published {@link LocalTrack}
    * @param {LocalTrack} track - the {@link LocalTrack}
    * @param {function(LocalTrackPublication): void} unpublish - The callback
    *   that unpublishes the {@link LocalTrackPublication}
    * @param {LocalTrackPublicationOptions} options - {@link LocalTrackPublication}
    *   options
    */
-  constructor(sid, track, unpublish, options) {
-    options = Object.assign({
-      logLevel: DEFAULT_LOG_LEVEL
-    }, options);
+  constructor(trackSid, track, unpublish, options) {
+    const isEnabled = typeof track.isEnabled === 'boolean'
+      ? track.isEnabled
+      : true;
 
-    const logLevels = buildLogLevels(options.logLevel);
+    super(isEnabled, track.name, options);
 
     Object.defineProperties(this, {
-      _instanceId: {
-        value: nInstances++
-      },
-      _log: {
-        value: options.log || new Log('default', this, logLevels),
+      _onTrackEvent: {
+        value: () => {
+          this.isTrackEnabled = track.isEnabled;
+        }
```

Comment:
Can you move this out of `defineProperties` and do, e.g.,

```
class Foo {
  bar = () => {
    // ...
  }
}
```

There should be a Babel plugin to allow this.